CentOS 7: fatal error: linux/kcmp.h: No such file or directory

# Required information

 * Distribution: CentOS
 * Distribution Version: 7.7.1908
 * Architecture: x86_64

Build environment:
 * golang: 1.13.3-1.el7
 * gcc-4.8.5-39.el7
 * glibc-devel: 2.17-292.el7
 * kernel-headers: 3.10.0-1062.9.1.el7
 * libcap-ng-0.7.5-4.el7
 * libseccomp-devel-2.3.1-3.el7
 * lxc-devel: 3.2.1-0.3.el7

# Issue description

Building LXD >=3.19 fails with the following error:
```
[...]
cd /builddir/build/BUILD/lxd-3.20/_output/src/github.com/lxc/lxd/lxd
pkg-config --cflags -- lxc libcap
pkg-config --libs -- lxc libcap
CGO_LDFLAGS='"-L/builddir/build/BUILD/lxd-3.20/_dist/deps/sqlite/.libs/" "-L/builddir/build/BUILD/lxd-3.20/_dist/deps/libco/" "-L/builddir/build/BUILD/lxd-3.20/_dist/deps/raft/.libs/" "-L/builddir/build/BUILD/lxd-3.20/_dist/deps/dqlite/.libs/" "-Wl,-rpath,/usr/lib64/lxd" "-llxc" "-lutil" "-lcap"' /usr/lib/golang/pkg/tool/linux_amd64/cgo -objdir $WORK/b001/ -importpath github.com/lxc/lxd/lxd -- -I $WORK/b001/ -I/builddir/build/BUILD/lxd-3.20/_dist/deps/sqlite/ -I/builddir/build/BUILD/lxd-3.20/_dist/deps/libco/ -I/builddir/build/BUILD/lxd-3.20/_dist/deps/raft/include/ -I/builddir/build/BUILD/lxd-3.20/_dist/deps/dqlite/include/ -std=gnu11 -Wvla -Werror -fvisibility=hidden -Winit-self -Wformat=2 -Wshadow -Wendif-labels -fasynchronous-unwind-tables -pipe --param=ssp-buffer-size=4 -g -Wunused -Werror=implicit-function-declaration -Werror=return-type -Wendif-labels -Werror=overflow -Wnested-externs -fexceptions ./cgo.go ./devices.go ./main_checkfeature.go ./main_forkfile.go ./main_forkmount.go ./main_forknet.go ./main_forkproxy.go ./main_forksyscall.go ./main_forkuevent.go ./main_nsexec.go
# github.com/lxc/lxd/lxd
_output/src/github.com/lxc/lxd/lxd/main_checkfeature.go:13:24: fatal error: linux/kcmp.h: No such file or directory
 #include <linux/kcmp.h>
                        ^
compilation terminated.
```

The default CentOS 7 kernel-headers package doesn't contain `linux/kcmp.h`. Building with a newer `kernel-headers-4.4.212-1.el7.elrepo` is successful.

I tried to check what this header is used for by removing the failing include which then results in a missing `KCMP_FILE` declaration:
```
# github.com/lxc/lxd/lxd
_output/src/github.com/lxc/lxd/lxd/main_checkfeature.go: In function 'filecmp':
_output/src/github.com/lxc/lxd/lxd/main_checkfeature.go:162:40: error: 'KCMP_FILE' undeclared (first use in this function)
  return syscall(__NR_kcmp, pid1, pid2, KCMP_FILE, fd1, fd2);
                                        ^
_output/src/github.com/lxc/lxd/lxd/main_checkfeature.go:162:40: note: each undeclared identifier is reported only once for each function it appears in
```

I'm reporting this, as per [this comment](https://discuss.linuxcontainers.org/t/lxd-3-19-on-centos-7-via-copr/6696/4) there might be some way to solve this so that LXD could be build using the stock headers?

--- a/lxd/main_checkfeature.go
+++ b/lxd/main_checkfeature.go
@@ -10,7 +10,6 @@ import (
 #endif
 #include <errno.h>
 #include <fcntl.h>
-#include <linux/kcmp.h>
 #include <linux/types.h>
 #include <poll.h>
 #include <stdbool.h>
@@ -161,7 +160,10 @@ static int user_trap_syscall(int nr, unsigned int flags)
 static int filecmp(pid_t pid1, pid_t pid2, int fd1, int fd2)
 {
 #ifdef __NR_kcmp
-	return syscall(__NR_kcmp, pid1, pid2, KCMP_FILE, fd1, fd2);
+#ifndef KCMP_FILE
+#define KCMP_FILE 0
+#endif
+	return syscall(__NR_kcmp, pid1, pid2, 0, fd1, fd2);
 #else
 	errno = ENOSYS;
 	return -1;

